### ISSUE TITLE
Migration error Thread.currentThread().contextClassLoader must be set

### ISSUE THREAD

**Issue by leo-lb** [etesync/android#157]
Hello!

After logging in/signing up and pressing "Create" after ticking all items to migrate, the migration seems to go on well and then just before finishing it throws this error:

![Screenshot_20201129-154704_EteSync_1](https://user-images.githubusercontent.com/44934014/100545089-60f44a00-325a-11eb-8103-f9d6e6830b56.png)

I've tried a few times, restarting the app etc. no luck.

Version: 2.1.5 from FDroid

Thank you

**Comment by tasn** [comment]
This is very weird, and it'd be hard to figure out what's going on without a backtrace. Can you by any chance turn on logging to file from the settings and attach the relevant backtrace here (if you only copy the backtrace it won't include any private information).

Btw, if it throws this towards the end, to the point that you think it actually finished, you can probably just try to add your EteSync 2.0 account to the Android app and see if it actually finished migrating.

**Comment by leo-lb** [comment]
@tasn I just restarted the app *again* and it went OK. Not sure I will be able to reproduce.

**Comment by leo-lb** [comment]
@tasn Otherwise, will we get a fully transparent migration process one day? I feel uneasy with the current one, it creates other contact stores and everything in my phone which will break my current configuration. It would be nice if the migration could re-use the existing contact stores so that I don't have any configuration change with my phone to make. I remember struggling with that and duplicating stuff across my contact stores which I thankfully solved but I don't want to go through it again, it has lots of friction.

**Comment by tasn** [comment]
I'm not sure what you mean by transparent migration. Though if you are referring to the fact that it creates new address books and etc? Unfortunately that's the only way it can be done. The reason for that is that the EteSync 2.0 is actually a new account, so as far as Android is concerned, this is completely different data.
I don't think we will ever do another migration like this. We went by four years until we did this one, and it's definitely painful so we will avoid it as much as possible.

I'm closing this ticket as the issue can't be reproduced. Though I'll definitely keep an eye out for this happening to anyone else. Thanks again for reporting!
